src/robot.coffee

```diff
@@ -116,7 +125,7 @@ class Robot
         modifiers
       )
 
-    @listeners.push new TextListener(@, newRegex, options, callback)
+    return newRegex
```

Comment:
According @technicalpickles 's feedback on https://github.com/github/hubot/pull/922 the return statements should be implicit.